src/configuration.rs

```diff
@@ -0,0 +1,128 @@
+//! Types for managing and extracting configuration.
+
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
```

Comment:
A possible performance improvement here would be to use the [hashbrown](https://github.com/Amanieu/hashbrown) crate.

Reply:
AFAICT hashbrown is used only in rustc, not libstd.

Reply:
Oh I missed this. But yeah, it's landing in libstd; people made a PR during RustFest Rome for this!